common/springUtilities/unitScripts.lua

```diff
@@ -0,0 +1,420 @@
+-- Various helpers for bridging the gap in unit data between defs and scripts.
+
+local function removeCobComments(content)
+	local n = #content
+	local i = 1
+	local output = {}
+
+	local escape = "\\"
+	local line_feed = "\n"
+	local line_return = "\r"
+	local line_ends = { [line_feed] = true, [line_return] = true }
+	local quotes = { ['"'] = true, ["'"] = true }
+
+	local insert = table.insert
+	local function current() return string.sub(content, i, i) end
+	local function starts_with(s)
+		return string.sub(content, i, i + #s - 1) == s
+	end
+
+	while i <= n do
+		local ch = current()
+
+		if quotes[ch] then
+			-- Literals
+			local quote = ch
+			insert(output, quote)
+			i = i + 1
+			while i <= n do
+				local c = current()
+				insert(output, c)
+				i = i + 1
+				if c == escape then
+					if i <= n then
+						insert(output, current())
+						i = i + 1
+					end
+				elseif c == quote then
+					break
+				end
+			end
+
+		elseif starts_with("//") then
+			-- Single-line comment
+			i = i + 2
+			while i <= n do
+				local c = current()
+				i = i + 1
+				if c == line_return then
+					insert(output, line_return)
+					if i <= n and current() == line_feed then
+						insert(output, line_feed)
+						i = i + 1
+					end
+					break
+				elseif c == line_feed then
+					insert(output, line_feed)
+					break
+				end
+			end
+
+		elseif starts_with("/*") then
+			-- Multi-line comment
+			i = i + 2
+			while i <= n do
+				if starts_with("*/") then
+					i = i + 2
+					break
+				else
+					local c = current()
+					if line_ends[c] then
+						insert(output, c)
+					end
+					i = i + 1
+				end
+			end
+		else
+			insert(output, ch)
+			i = i + 1
+		end
+	end
+
+	return table.concat(output)
+end
+
+local function removeLusComments(content)
```

Comment:
there's some sort of lua parser at https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/common/springUtilities/safeluaparser.lua

would be good to reuse it